### ISSUE TITLE
Bookies supports ephemeral port 

### ISSUE THREAD

**Issue by zhaijack** [apache/bookkeeper#296]
**FEATURE REQUEST**

1. Please describe the feature you are requesting.

            // If the caller specified ephemeral ports then use ephemeral ports for all
            // the bookies else use numBookie ports starting at initialPort
            if (0 == initialPort) {
                bsConfs[i].setBookiePort(0);
            } else {
                bsConfs[i].setBookiePort(initialPort + i);
            }

Currently there is a logic in LocalBookKeeper to support ephemeral ports for testing, however the corresponding logic in Bookie is missing. Somehow DL uses this feature for testing, it would be nice to port the ephemeral port supports.

2. Indicate the importance of this issue to you (blocker, must-have, should-have, nice-to-have). Are you currently using any workarounds to address this issue?

**nice-to-have**. marked it as **nice-to-have**, but it might be **blocker** for DL to use latest BK. 

3. Provide any additional detail on your proposed use case for this feature.

**Comment by eolivelli** [comment]
I think that for production it is not possible to use an ephemeral port because actually the port is part of the ID of the bookie.
Some time ago I filed this issue to make the bookie id totally customizable
https://issues.apache.org/jira/browse/BOOKKEEPER-875

the implementation is not simple, because we use the ID both as an "ID" and an "Address"

**Comment by sijie** [comment]
@eolivelli : I think the descriptions says it is only for testing, and it is typically a good practice to use ephemeral ports for testing, it can avoid port conflicts and allow executing tests in parallel.

**Comment by eolivelli** [comment]
Yes I just wanted to make it clear.
For me is ok
